### PR TITLE
Print inifile and rootdir when there's usage errors

### PR DESCRIPTION
**Pull request by nicoddemus**
Related to #821

### CODE REVIEW

File: _pytest/config.py [hunk 1]
```diff
@@ -669,10 +674,13 @@ def _addoption_instance(self, option, shortupper=False):
 
 
 class MyOptionParser(argparse.ArgumentParser):
-    def __init__(self, parser):
+    def __init__(self, parser, extra_info={}):
```
Comment (joshmoore) [inline _pytest/config.py]: Likely not an issue but: 
http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

**Comment by asfaltboy** [comment]
:+1:  nice!

File: _pytest/config.py (hunk 1 above)
Comment (hpk42) [inline _pytest/config.py]: it should indeed be "extra_info = None" on principle grounds.

Comment (nicoddemus) [inline _pytest/config.py]: Agreed, thanks

**Comment by nicoddemus** [comment]
Removed the mutable dict from the parameter list as requested.

**Comment by nicoddemus** [comment]
Rebased on master to account for fix in py26.

**Pull request merged**